op-mode-definitions/poweroff.xml

```diff
@@ -2,37 +2,45 @@
 <interfaceDefinition>
   <node name="poweroff">
     <properties>
-      <help>Poweroff the system</help>
+      <help>Shutdown the system</help>
     </properties>
-    <command>/opt/vyatta/bin/sudo-users/vyatta-poweroff.pl --action poweroff</command>
+    <command>sudo ${vyos_op_scripts_dir}/powerctrl.py --poweroff now</command>
+
     <children>
       <leafNode name="now">
         <properties>
-          <help>Poweroff the system without confirmation</help>
+          <help>Shutdown the system imidiently</help>
         </properties>
-        <command>/opt/vyatta/bin/sudo-users/vyatta-poweroff.pl --action poweroff --now</command>
+        <command>sudo ${vyos_op_scripts_dir}/powerctrl.py --yes --poweroff now</command>
       </leafNode>
-      
+
       <leafNode name="cancel">
         <properties>
-          <help>Cancel a pending poweroff</help>
+          <help>Cancel a pending reboot</help>
         </properties>
-        <command>/opt/vyatta/bin/sudo-users/vyatta-poweroff.pl --action poweroff_cancel</command>
+        <command>sudo ${vyos_op_scripts_dir}/powerctrl.py --cancel</command>
       </leafNode>
 
+    <tagNode name="in">
+      <properties>
+        <help>Reboot in X minutes</help>
+        <completionHelp>
+          <list>Minutes</list>
+        </completionHelp>
+      </properties>
+      <command>sudo ${vyos_op_scripts_dir}/powerctrl.py --yes --poweroff '$3'</command>
+
+    </tagNode>
 
     <tagNode name="at">
       <properties>
-        <help>Poweroff at a specific time</help>
+        <help>Reboot at a specific time</help>
         <completionHelp>
           <list>HH:MM</list>
-          <list>MMDDYY</list>
```

Comment:
Please keep this completion as it was present in VyOS 1.1.x.

Reply:
The problem with this is that systemd-shutdownd doesent allow us to specify dates directly, an implementation of this could be added but it need to be converted into minutes until this date.   i'll try to implement it.